packages/replay/src/index.ts

```diff
@@ -234,1119 +126,35 @@ export class Replay implements Integration {
-    // completing) into a single flush.
-
-    try {
-      await this.flushLock;
-    } catch (err) {
-      __DEBUG_BUILD__ && logger.error(err);
-    } finally {
-      this.debouncedFlush();
-    }
-  };
-
-  /**
-   *
-   * Always flush via `debouncedFlush` so that we do not have flushes triggered
-   * from calling both `flush` and `debouncedFlush`. Otherwise, there could be
-   * cases of mulitple flushes happening closely together.
-   */
-  flushImmediate(): Promise<void> {
-    this.debouncedFlush();
-    // `.flush` is provided by lodash.debounce
-    return this.debouncedFlush.flush();
-  }
-
-  /**
-   * Send replay attachment using `fetch()`
-   */
-  async sendReplayRequest({
-    events,
-    replayId: event_id,
-    segmentId: segment_id,
-    includeReplayStartTimestamp,
-    eventContext,
-  }: SendReplay): Promise<void | undefined> {
-    const payloadWithSequence = createPayload({
-      events,
-      headers: {
-        segment_id,
-      },
-    });
-
-    const { urls, errorIds, traceIds, initialTimestamp } = eventContext;
-
-    const currentTimestamp = new Date().getTime();
-
-    const sdkInfo = {
-      name: 'sentry.javascript.integration.replay',
-      version: __SENTRY_REPLAY_VERSION__,
-    };
-
-    const replayEvent = await new Promise(resolve => {
-      getCurrentHub()
-        // @ts-ignore private api
-        ?._withClient(async (client: Client, scope: Scope) => {
-          // XXX: This event does not trigger `beforeSend` in SDK
-          // @ts-ignore private api
-          const preparedEvent: Event = await client._prepareEvent(
-            {
-              type: REPLAY_EVENT_NAME,
-              ...(includeReplayStartTimestamp ? { replay_start_timestamp: initialTimestamp / 1000 } : {}),
-              timestamp: currentTimestamp / 1000,
-              error_ids: errorIds,
-              trace_ids: traceIds,
-              urls,
-              replay_id: event_id,
-              segment_id,
-            },
-            { event_id },
-            scope,
-          );
-          const session = scope && scope.getSession();
-          if (session) {
-            // @ts-ignore private api
-            client._updateSessionFromEvent(session, preparedEvent);
-          }
-
-          preparedEvent.sdk = {
-            ...preparedEvent.sdk,
-            ...sdkInfo,
-          };
-
-          preparedEvent.tags = {
-            ...preparedEvent.tags,
-            sessionSampleRate: this.options.sessionSampleRate,
-            errorSampleRate: this.options.errorSampleRate,
-            replayType: this.session?.sampled,
-          };
+  private _setup(): void {
+    // Client is not available in constructor, so we need to wait until setupOnce
+    this._loadReplayOptionsFromClient();
 
-          resolve(preparedEvent);
-        });
+    const replay = new ReplayContainer({
+      options: this.options,
+      recordingOptions: this.recordingOptions,
```

Comment:
is this intermediate var needed?

```suggestion
    this.replay = new ReplayContainer({
      options: this.options,
      recordingOptions: this.recordingOptions,
```

Reply:
no, actually not 👍